### ISSUE TITLE
TakePhotoAsync: Attempt to invoke virtual method 'android.content.res.XmlResourceParser 

### ISSUE THREAD

**Issue by nikkilocke** [jamesmontemagno/MediaPlugin#381]
## Bug Information

Version Number of Plugin: 3.0.1
Device Tested On: Google Nexus 10
Simulator Tested On:
Version of VS: 2017 Community 15.4.1
Version of Xamarin: 4.7.10.22, Android SDK 8.0.0.33

### Steps to reproduce the Behavior

In App.xaml.cs:

		protected async override void OnStart ()
		{
			// Handle when your app starts
			try {
				var cameraStatus = await CrossPermissions.Current.CheckPermissionStatusAsync(Permission.Camera);

				if (cameraStatus != PermissionStatus.Granted) {
					var results = await CrossPermissions.Current.RequestPermissionsAsync(new[] { Permission.Camera });
					cameraStatus = results[Permission.Camera];
				}

				if (cameraStatus == PermissionStatus.Granted) {
					await Plugin.Media.CrossMedia.Current.Initialize();
                    CanUseCamera = Plugin.Media.CrossMedia.Current.IsCameraAvailable && Plugin.Media.CrossMedia.Current.IsTakePhotoSupported;
				}
			} catch(Exception ex) {
                Utils.Report(ex);
			}
		}

In an ordinary page:

        private async void Photo_Clicked(object sender, EventArgs e) {
            try {
                using (var photo = await Plugin.Media.CrossMedia.Current.TakePhotoAsync(new Plugin.Media.Abstractions.StoreCameraMediaOptions() {
                    CompressionQuality = 0,
                    PhotoSize = Plugin.Media.Abstractions.PhotoSize.Medium
                })) {
                    if (photo != null) {
                        Image = new MemoryStream();
                        using (Stream s = photo.GetStream())
                            s.CopyTo(Image);
                        Image.Position = 0;
                        PhotoImage.Source = ImageSource.FromStream(() => Image);
                    }
                }
            } catch (Exception ex) {
                Utils.Report(ex);
                Message.Text = ex.Message;
            }
        }

### Expected Behavior

Photo is taken and stored.

### Actual Behavior

The call to TakePhoto throws an exception:

	Java.Lang.RuntimeException: Attempt to invoke virtual method 'android.content.res.XmlResourceParser android.content.pm.ProviderInfo.loadXmlMetaData(android.content.pm.PackageManager, java.lang.String)' on a null object reference
	  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <5830ce0de8e942bbbd70870f4c2f1f06>:0 
	  at Java.Interop.JniEnvironment+StaticMethods.CallStaticObjectMethod (Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x00069] in <2bc43a37b3cd4070a78772810a2628cb>:0 
	  at Android.Runtime.JNIEnv.CallStaticObjectMethod (System.IntPtr jclass, System.IntPtr jmethod, Android.Runtime.JValue* parms) [0x0000e] in <4d61b611aef448eeab01db0e1d363ffe>:0 
	  at Android.Support.V4.Content.FileProvider.GetUriForFile (Android.Content.Context context, System.String authority, Java.IO.File file) [0x00077] in <0a46c0438f5e4c2d8218af82d400408d>:0 
	  at Plugin.Media.MediaPickerActivity.OnCreate (Android.OS.Bundle savedInstanceState) [0x0023f] in C:\projects\mediaplugin\src\Media.Plugin.Android\MediaPickerActivity.cs:162 
	--- End of stack trace from previous location where exception was thrown ---
	  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <5830ce0de8e942bbbd70870f4c2f1f06>:0 
	  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <5830ce0de8e942bbbd70870f4c2f1f06>:0 
	  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <5830ce0de8e942bbbd70870f4c2f1f06>:0 
	  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <5830ce0de8e942bbbd70870f4c2f1f06>:0 
	  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <5830ce0de8e942bbbd70870f4c2f1f06>:0 
	  at Plugin.Media.MediaImplementation+<TakePhotoAsync>d__16.MoveNext () [0x000c7] in C:\projects\mediaplugin\src\Media.Plugin.Android\MediaImplementation.cs:119 
	--- End of stack trace from previous location where exception was thrown ---
	  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <5830ce0de8e942bbbd70870f4c2f1f06>:0 
	  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <5830ce0de8e942bbbd70870f4c2f1f06>:0 
	  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <5830ce0de8e942bbbd70870f4c2f1f06>:0 
	  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <5830ce0de8e942bbbd70870f4c2f1f06>:0 
	  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <5830ce0de8e942bbbd70870f4c2f1f06>:0 
	  at DefectReport.DefectPage+<Photo_Clicked>d__5.MoveNext () [0x00048] in C:\Dev\DefectReport\DefectReport\DefectReport\DefectPage.xaml.cs:39 
	  --- End of managed Java.Lang.RuntimeException stack trace ---
	java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.res.XmlResourceParser android.content.pm.ProviderInfo.loadXmlMetaData(android.content.pm.PackageManager, java.lang.String)' on a null object reference
		at android.support.v4.content.FileProvider.parsePathStrategy(FileProvider.java:584)
		at android.support.v4.content.FileProvider.getPathStrategy(FileProvider.java:558)
		at android.support.v4.content.FileProvider.getUriForFile(FileProvider.java:400)
		at md5f1b52d9f1d5d1981437e7e130649468a.MediaPickerActivity.n_onCreate(Native Method)
		at md5f1b52d9f1d5d1981437e7e130649468a.MediaPickerActivity.onCreate(MediaPickerActivity.java:42)
		at android.app.Activity.performCreate(Activity.java:5990)
		at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1106)
		at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2278)
		at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2387)
		at android.app.ActivityThread.access$800(ActivityThread.java:151)
		at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1303)
		at android.os.Handler.dispatchMessage(Handler.java:102)
		at android.os.Looper.loop(Looper.java:135)
		at android.app.ActivityThread.main(ActivityThread.java:5254)
		at java.lang.reflect.Method.invoke(Native Method)
		at java.lang.reflect.Method.invoke(Method.java:372)
		at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
		at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)

**Comment by nikkilocke** [comment]
Just tested this on a Nexus 4 phone, and it does the same thing, so it's probably something I'm doing wrong.

**Comment by jamesmontemagno** [comment]
You must fill in your package name follow: https://github.com/jamesmontemagno/MediaPlugin/blob/master/README.md#android

**Comment by nikkilocke** [comment]
Thanks again, James. Just so you know I wasn't being completely thick, I had gone into Android Project/Properties/Android Manifest, and set the package name from "DefectReport.Android" to "DefectReport". I used that package name in the AndroidManifest.xml file, exactly as instructed. But, now I go back into the project properties, the package name has been changed to "DefectReport.DefectReport". I have no idea why.
